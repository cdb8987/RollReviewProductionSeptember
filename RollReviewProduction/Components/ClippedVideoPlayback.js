import * as React from 'react';
import { View, StyleSheet, Image, TouchableOpacity, Text } from 'react-native';
import { Video, Audio, ResizeMode } from 'expo-av';
import {useEffect, useState} from 'react'
import { Divider} from 'react-native-paper';




export default function ClippedVideoPlayback({
  videoRecord, 
  isFocused, 
  duration, 
  selectedVideoKey, 
  onSelect}) {
  //inFocusStatus refers to whether the video is scrolled over and hovered on
  

  const videoRef = React.useRef(null);
  const [status, setStatus] = useState({});

  const triggerAudio = async (videoRef) => {
  await Audio.setAudioModeAsync({ playsInSilentModeIOS: true });
  videoRef.current.playAsync();
  };

  
  useEffect(() => {
    if (status.isPlaying){triggerAudio(videoRef)}
  }, [videoRef, status.isPlaying])




  const endTimeStamp = videoRecord.timestamp>10000? videoRecord.timestamp-10000 + duration : videoRecord.timestamp + duration
  const dateArray=['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday']
  const dateRecord = new Date(videoRecord.createdAt)
  const day = dateArray[dateRecord.getDay()]
  const date = dateRecord.getDate()
  const month = dateRecord.getMonth()+1
  const year = dateRecord.getFullYear()

  return isFocused && selectedVideoKey===videoRecord.recordID?
    <>
    <TouchableOpacity 
    style={styles.container}
    onPress={()=>{
      status.isPlaying? videoRef.current.pauseAsync() : videoRef.current?.playAsync()
    }}>
      <View>
        <Text style={styles.feedText}>{videoRecord.technique} | {videoRecord.position} | {videoRecord.result}</Text>
        <Text >{[day, ' ', month,'-',date, ' ', year ]}</Text>
        <Text>NOTES: {videoRecord.notes}</Text>
        <Text>VIDEO URI{videoRecord.uri}</Text>
        
      </View> 
      <Video
        ref={videoRef}
        style={styles.video}
        source={{uri: videoRecord.uri}}
        // useNativeControls={true}
        positionMillis={videoRecord.timestamp>10000? videoRecord.timestamp -10000: 0}
        resizeMode={ResizeMode.COVER}
        isLooping
        shouldPlay={true}
        onPlaybackStatusUpdate={(status) => { 
            if(status.positionMillis > endTimeStamp){
                const handleVideoEnd = async()=>{
                await videoRef.current.setPositionAsync(videoRecord.timestamp - 10000)

                }
                handleVideoEnd()
        
                
                // videoRef.current.pauseAsync()
            }
            return setStatus(() => status)
        }
        }
      />
      {/* <View style={styles.buttons}>
        <Button
          title={status.isPlaying ? 'Pause' : 'Play'}
          onPress={() =>
            status.isPlaying ? video.current.pauseAsync() : video.current.playAsync()
          }
        />
      </View> */}
      <Divider/>
    </TouchableOpacity>
    </>
  : 

  <>
  <TouchableOpacity
    style={styles.container}
    onPress={()=>{ onSelect()}}
    
    
    >
    <View><Text style={styles.feedText}>{videoRecord.technique} | {videoRecord.position} | {videoRecord.result}</Text><Text >{[day, ' ', month,'-',date, ' ', year ]}</Text><Text>NOTES: {videoRecord.notes}</Text></View>
       
  
  {/* Image will have to be tailored to IOS and Android separately.  Use https://reactnative.dev/docs/images to correc this later */}
  {/* Android */}
  <Image
  source={{uri: `${videoRecord.thumbnailURI}`}}
  style={styles.video}
/>

  {/* IOS */}
  {/* <Image
  source={{uri: 'app_icon'}}
  style={{width: 40, height: 40}}
/> */}
  
  </TouchableOpacity>
  <Divider/>
  </>


}

const styles = StyleSheet.create({
    container:{flex: 1, justifyContent: 'center', alignItems: 'center', paddingTop: 10},
    buttons:{flex: 1},
    video: {height: 300, width: 300, justifyContent: 'center', alignItems: 'center'}, 
    feedText:{
      fontSize:24, 
      fontWeight: 'bold'
    }
})