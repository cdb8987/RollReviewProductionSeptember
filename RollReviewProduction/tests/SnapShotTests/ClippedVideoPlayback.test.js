import React from "react";
import renderer from 'react-test-renderer';
import ClippedVideoPlayback from '../../Components/ClippedVideoPlayback';

test('renders ClippedVideoPlayback snapshot', () => {
    const timeStamp= new Date()
    const tree = renderer.create(<ClippedVideoPlayback 

  videoRecord={{"result": "Improved Position", "technique": "Escape", "thumbnailURI": "file:///data/user/0/host.exp.exponent/files/1695665429971.jpg", "timestamp": 998, "uri": "file:///data/user/0/host.exp.exponent/files/1695648599547.mp4"}}
  
  
  isFocused={true}
  duration={0}
  selectedVideoKey={0}
  onSelect={()=>{null}}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});