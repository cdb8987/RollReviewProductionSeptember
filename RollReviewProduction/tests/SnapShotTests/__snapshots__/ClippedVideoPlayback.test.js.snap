// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ClippedVideoPlayback snapshot 1`] = `
[
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "paddingTop": 10,
      }
    }
  >
    <View>
      <Text
        style={
          {
            "fontSize": 24,
            "fontWeight": "bold",
          }
        }
      >
        Escape
         | 
         | 
        Improved Position
      </Text>
      <Text>
         
        NaN
        -
        NaN
         
        NaN
      </Text>
      <Text>
        NOTES: 
      </Text>
    </View>
    <Image
      source={
        {
          "uri": "file:///data/user/0/host.exp.exponent/files/1695665429971.jpg",
        }
      }
      style={
        {
          "alignItems": "center",
          "height": 300,
          "justifyContent": "center",
          "width": 300,
        }
      }
    />
  </View>,
  <View
    style={
      [
        {
          "backgroundColor": "rgba(202, 196, 208, 1)",
          "height": 0.5,
        },
        undefined,
        false,
        false,
        undefined,
      ]
    }
  />,
]
`;
